packages/react-components/src/shared/src/useMergedRefs.ts

```diff
@@ -11,24 +11,22 @@ export function assignRef<T>(ref: ForwardedRef<T>, node: T): void {
     }
 }
 
-export function mergeRefs<T>(...refs: ForwardedRef<T>[]): (current: T) => void {
-    const mergedRef = (current: T): void => {
-        // Resulting callback function can be used has a ref.
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore
-        mergedRef.current = current;
+type MergedRef<T> = ((instance: T | null) => void) & MutableRefObject<T | null>;
 
+export function mergeRefs<T>(...refs: ForwardedRef<T>[]): MergedRef<T> {
+    const mergedRef: MergedRef<T> = ((current: T): void => {
+        mergedRef.current = current;
         refs
             .filter(Boolean)
             .forEach(ref => {
                 assignRef(ref, current);
             });
-    };
+    }) as MergedRef<T>;
```

Comment:
so we specify in `mergeRef` declaration that it's a MergedRef<T> and we also need a cast here?

Maybe we should remove those 2 declaration and only do `return mergedRef as MergedRef<T>` ? Would it work?

Reply:
i succesfully removed on of the MergeRef.

This scenario is tricky since a MergeRef has the current property, but in our case the current property is set within the function.